### PR TITLE
Adding analytics to the demo

### PR DESCRIPTION
**Pull request by gauntface**
@jaffathecake, @crhyme and @surma would super appreciate your eyes on `src/scripts/analytics.js` and `src/service-worker.js` changes.

### CODE REVIEW

**Comment by surma** [comment]
*cough* @jakearchibald @crhym3 *cough*

**Comment by gauntface** [comment]
I blame github not auto-completing all users - p.s. thank you @surma :)

**Comment by gauntface** [comment]
Side note: Travis is actually running and passing tests - it's just failing on Firefox Beta which has a problem with Webdriver at the moment.

File: src/scripts/analytics.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        method: 'post',
+        body: payloadString
+      });
+    })
+    .then(response => {
+      if (response.status !== 200) {
+        return response.text()
+        .then(responseText => {
+          throw new Error(
+            `Bad response from Google Analytics ` +
```
Comment (crhym3) [inline src/scripts/analytics.js]: Personally, I wouldn't even bother: https://developers.google.com/analytics/devguides/collection/protocol/v1/reference#response-codes says

> The Measurement Protocol **does not return an error** code if the payload data was malformed, or if the data in the payload was incorrect or was not processed by Google Analytics

File: src/service-worker.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
 'use strict';
 
-/* eslint-env serviceworker */
+/* eslint-env browser, serviceworker */
+
+importScripts('./scripts/analytics.js');
 
 self.addEventListener('push', function(event) {
+  self.analytics.trackEvent('push-received');
```
Comment (crhym3) [inline src/service-worker.js]: aren't you missing `eventValue` arg here?

Comment (gauntface) [inline src/service-worker.js]: All values are filtered out before being passed to analytics in the payload. The eventAction is the only required field.

File: src/scripts/analytics.js (hunk 1 above)
Comment (gauntface) [inline src/scripts/analytics.js]: Still useful if ad blocker stops the request from going through.

File: src/service-worker.js (hunk 2 above)
Comment (crhym3) [inline src/service-worker.js]: Well, that's not what src/scripts/analytics.js says :)

```js
 +    if (!eventAction && !eventValue) {
 +      console.warn('sendAnalyticsEvent() called with no eventAction or ' +
 +      'eventValue.');
 +      return Promise.resolve();
 +    }
```

Comment (crhym3) [inline src/service-worker.js]: sorry, ignore this!

**Comment by crhym3** [comment]
:+1:

File: src/service-worker.js (hunk 2 above)
Comment (gauntface) [inline src/service-worker.js]: Nah you are right - I just want to enforce something to be defined is all.

**Pull request merged**